src/llmescache/langchain.py

```diff
@@ -0,0 +1,90 @@
+import hashlib
+from datetime import datetime
+from typing import Any, Optional
+
+import elasticsearch
+from langchain_core.caches import RETURN_VAL_TYPE, BaseCache
+from langchain_core.load import dumps
+
+
+class ElasticSearchCache(BaseCache):
+    """Cache store for LLM using ElasticSearch."""
+
+    def __init__(
+        self,
+        es_client: elasticsearch.Elasticsearch,
+        es_index: str,
+        store_input: bool = True,
+        store_datetime: bool = True,
+        additional_metadata: Optional[dict] = None,
+    ):
+        """
+        Initialize the ElasticSearch cache store.
+
+        Parameters:
+            es_client: Elasticsearch
+                The Elasticsearch client to use for the cache store.
+            es_index: str
+                The name of the index to use for the cache store.
+            store_input: bool
+                Whether to store the input in the cache, i.e. the LLM input messages.
```

Comment:
I would write
"Whether to store the LLM input in the cache, i.e. the model parameters and the prompt"
This means that this parameter should determine if we store the llm_string (that is not stored by default in langchain)

Reply:
Not sure about this. This parameter is used to store `prompt` non `llm_strings` from langchain.  The parameters are storend inside `llm_strings` while the input messages in `prompt`.

Reply:
ok I meant to consider everything related to the input as one type of information we could store or not. let's add another prameter `store_params` that allows to store the llm_string in field llm_params

Reply:
changed